File: cardstack/src/main/java/com/mutualmobile/cardstack/CardFrameLayout.java
package com.mutualmobile.cardstack;

import android.annotation.SuppressLint;
import android.widget.FrameLayout;

/**
 * Created by tushar on 7/18/16.
 */
@SuppressLint("ViewConstructor")
class CardFrameLayout extends FrameLayout {
    private final CardStackLayout parent;

    public CardFrameLayout(CardStackLayout cardStackLayout) {
        super(cardStackLayout.getContext());
        this.parent = cardStackLayout;
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        CardStackAdapter mAdapter = parent.getAdapter();
        if (mAdapter != null) {
            int height = (int) (mAdapter.getFullCardHeight() + (mAdapter.getCount() - 1) * parent.getCardGap());
            heightMeasureSpec = MeasureSpec.makeMeasureSpec(height, MeasureSpec.EXACTLY);
        }

        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
    }
}

File: cardstack/src/main/java/com/mutualmobile/cardstack/CardFrameLayout.kt
package com.mutualmobile.cardstack

import android.annotation.SuppressLint
import android.view.View
import android.widget.FrameLayout

/**
 * Created by tushar on 7/18/16.
 */
@SuppressLint("ViewConstructor")
internal class CardFrameLayout(private val parent: CardStackLayout) : FrameLayout(parent.context) {

  override fun onMeasure(
    widthMeasureSpec: Int,
    heightMeasureSpec: Int
  ) {
    var heightMeasureSpecs = heightMeasureSpec
    val mAdapter = parent.adapter
    if (mAdapter != null) {
      val height = (mAdapter.fullCardHeight + (mAdapter.count - 1) * parent.cardGap).toInt()
      heightMeasureSpecs = View.MeasureSpec.makeMeasureSpec(height, View.MeasureSpec.EXACTLY)
    }

    super.onMeasure(widthMeasureSpec, heightMeasureSpecs)
  }
}